code/lib/core-common/src/presets.ts

```diff
@@ -98,15 +107,21 @@ export const resolveAddonName = (
   // serve it up correctly  when yarn pnp or pnpm is being used.
   // Vite will be broken in such cases, because it does not process absolute paths,
   // and it will try to import from the bare import, breaking in pnp/pnpm.
-  const absolutizeExport = (exportName: string) => {
-    return resolve(`${name}${exportName}`);
+  const absolutizeExport = (exportName: string, preferMJS: boolean) => {
+    const found = resolve(`${name}${exportName}`);
+
+    if (found) {
+      return preferMJS ? resolvePathToMjs(found) : found;
```

Comment:
What's the scenario where mjs would not be preferred?

Reply:
presets.js in node

Reply:
got it! thanks for explaining